test/sanity/validate-modules/main.py

```
@@ -1374,7 +1374,7 @@ def validate(self):
 
             # See if current version => deprecated.removed_in, ie, should be docs only
             if docs and 'deprecated' in docs and docs['deprecated'] is not None:
-                removed_in = docs.get('deprecated')['removed_in']
+                removed_in = StrictVersion(str(docs.get('deprecated')['removed_in']))
```

This does fix the float issue, but will result in a trackback instead of a proper error message if the `removed_in` value isn't a valid version number. There should be an error code for that.

---

This isn't a new problem of course, the previous code would exhibit the same behavior.

---

Since we don't log an error here, I'm not positive what to do when reporting an invalid version.  I'll have to play around with it.

---

Would that be caught by https://github.com/ansible/ansible/blob/devel/test/sanity/validate-modules/schema.py#L126

---

The problem is that this check happens before schema validations.  So if the value is invalid, the comparison can traceback.  We would need to catch it, and do something useful with that before hand.

I just don't know what is useful.  Do we just make `end_of_deprecation_should_be_docs_only = False` in that scenario, and let the schema validation fail?  That seems like the right thing to do here.

---

As long as the schema validation will report the issue, it should be fine to ignore the error.